x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/index.ts

```diff
@@ -492,7 +539,7 @@ export class AIAssistantKnowledgeBaseDataClient extends AIAssistantDataClient {
         page: 1,
         sortField: 'created_at',
         sortOrder: 'asc',
-        filter: `${userFilter}${` AND type:index`}`, // TODO: Support global tools (no user filter), and filter by space as well
+        filter: `${userFilter} AND type:index`,
```

Comment:
Note: Removed `TODO` as userFilter covers this case (and it always only queries for the current space).